lib/postgrex/extensions/polygon.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Postgrex.Extensions.Polygon do
+  @moduledoc false
+  import Postgrex.BinaryUtils, warn: false
+  use Postgrex.BinaryExtension, send: "poly_send"
+  alias Postgrex.Extensions.Polygon
+  alias Postgrex.Extensions.Point
+
+  def encode(_) do
+    quote location: :keep do
+      %Postgrex.Polygon{vertices: vertices} when is_list(vertices) ->
+        len = << length(vertices)::int32 >>
+        vert = vertices
+        |> Enum.map(fn(p) -> Point.encode_point(p, Postgrex.Polygon) end)
```

Comment:
Don't pipe here. We usually don't pipe if there is only a single pipe. If you feel the line is too long (80 chars is not a hard limit though) you can rewrite it like this: `Enum.map(vertices, &Point.encode_point(&1, Postgrex.Polygon))`.